src/services/item/plugins/action/index.ts

```diff
@@ -107,7 +109,46 @@ const plugin: FastifyPluginAsync<GraaspActionsOptions> = async (fastify, options
     },
   );
 
-  // get actions matching the given `id`
+  fastify.route<{ Params: IdParam; Body: { type: string; extra?: { [key: string]: unknown } } }>({
+    method: HttpMethod.POST,
+    url: '/:id/actions',
+    schema: postAction,
+    preHandler: fastify.fetchMemberInSession,
+    handler: async (request) => {
+      const {
+        member,
+        params: { id: itemId },
+        body: { type, extra = {} },
+      } = request;
+
+      // allow only from known hosts
+      if (!request.headers.origin) {
+        throw new CannotPostAction();
+      }
+      const originUrl = new URL(request.headers.origin);
+      if (
+        !Object.values(CLIENT_HOSTS)
+          .map(({ url }) => url.hostname)
+          .includes(originUrl.hostname)
+      ) {
+        throw new CannotPostAction();
+      }
+
+      return db.transaction(async (manager) => {
+        const repositories = buildRepositories(manager);
+        const item = await itemService.get(member, repositories, itemId);
+        await actionService.postMany(member, repositories, request, [
+          {
+            item,
+            type,
+            extra,
```

Comment:
Nope, because we have 
```js
    const completeActions = actions.map((a) => ({
      member,
      geolocation: geolocation ?? undefined,
      view,
      extra: {},
      ...a,
    }));
```
where `a` is the data above.